### ISSUE TITLE
Plot writing / saving argument

### ISSUE THREAD

**Issue by grantmcdermott** [grantmcdermott/tinyplot#125]
_(Following on from https://github.com/grantmcdermott/tinyplot/issues/121#issuecomment-1941993963)_

Something like:

```r
tinyplot(1:10, save = "myplot.png")
```
or

```r
tinyplot(1:10, file = "myplot.png")
```

Internally, this would need to open the appropriate graphics device based on the filename extension (here: `png()`) and call `dev.off()` right at the end.

We could potentially even support "in-memory" saving via `recordPlot()` if no extension is detected... although the interface might not be as clean. (Users would still have to call `replayPlot()` afterwards?)

**Comment by vincentarelbundock** [comment]
I kind of like this, because I'm too dumb to ever remember any of the graphics settings.

**Comment by zeileis** [comment]
I can see that this might be nifty. However, my feeling is that this opens a whole new can of worms that might lead to "feature creep"

- Because if we do it, we need to support at the very least `height` and `width` and then the question is whether these should also work if we don't `save` but just plot on an existing device.
- But then we probably also want `bg` and `pointsize`, possibly `res` or `family`.
- And once we're there we might also want to support all of the arguments of the devices like `png()`, `pdf()`, `svg()`, etc.
- And when we are there I think it would be cleaner to open the device rather than adding all this infrastructure to `tinyplot` just to save us from saying `dev.off()` at the end.

In short: Selecting the supported parameters is not straightforward IMO.

**Comment by grantmcdermott** [comment]
Thanks @zeileis, I definitely hear you on the risk of feature creep. But, motivated by the fact that I personally would find this useful, I think we can potentially strike a reasonable compromise for the most common use cases:

- Only support a subset of known devices (e.g., `png`, `pdf`, and `svg`)
- Set `res`, `width`, `height`, and `units` defaults via `tpar` (at 5, 8, "in", and 300 respectively) 
- etc.

Let me test out a dummy version and then submit a PR for you both to look at.

**Comment by zeileis** [comment]
OK, thanks, worth a try!

Will `height`/`width`/`units` only be honored when plotting to one of the file devices?

**Comment by J-Moravec** [comment]
Perhaps less obvious to users, but `tinyplot(..., dev = NULL)`

When dev is `NULL` default device is used, guessed in the same way it is done for standard `plot()`.

The `dev` can be either device from `?Devices`, or file name with an extension from which the device will be guestimated.

**Comment by grantmcdermott** [comment]
Hey folks, I have a PR supporting this in #143 (branch: ["filename"](https://github.com/grantmcdermott/tinyplot/tree/filename)). This is one of the last things I want to enable / decide on before CRAN submission. So please try it out and report back under the PR.
